webapp/channels/src/components/root/root.tsx

```diff
@@ -650,6 +650,23 @@ export default class Root extends React.PureComponent<Props, State> {
                         <TeamSidebar/>
                         <DelinquencyModalController/>
                         <Switch>
+                            {this.props.products?.filter((product) => product.publicComponent !== null).map((product) => (
```

Comment:
Mainly for readability

```suggestion
                            {this.props.products?.filter((product) => Boolean(product.publicComponent)).map((product) => (
```

Reply:
Addressed in [fab9018](https://github.com/mattermost/mattermost-server/pull/22710/commits/fab9018f0fd722c52d1075b4b0eff17be667b2a7)